modnet/featurizers/featurizers.py

```diff
@@ -77,9 +77,19 @@ def featurize(self, df: pd.DataFrame) -> pd.DataFrame:
             The featurized DataFrame.
 
         """
-        df_composition = self.featurize_composition(df)
-        df_structure = self.featurize_structure(df)
-        df_site = self.featurize_site(df)
+        if self.composition_featurizers or self.oxid_composition_featurizers:
+            df_composition = self.featurize_composition(df)
+        else:
+            df_composition = pd.DataFrame([])
+        if self.structure_featurizers:
+            df_structure = self.featurize_structure(df)
+        else:
+            df_structure = pd.DataFrame([])
+        if self.site_featurizers:
+            df_site = self.featurize_site(df)
+        else:
+            df_site = pd.DataFrame([])
```

Comment:
Minor tweak to remove some lines

```suggestion
        df_composition = pd.DataFrame([])
        if self.composition_featurizers or self.oxid_composition_featurizers:
            df_composition = self.featurize_composition(df)
            
        df_structure = pd.DataFrame([])            
        if self.structure_featurizers:
            df_structure = self.featurize_structure(df)
            
        df_site = pd.DataFrame([])
        if self.site_featurizers:
            df_site = self.featurize_site(df)           
```